src/selectors.js

```diff
@@ -41,51 +44,79 @@ export const withDenormalize = thunk =>
     }
   );
 
-export const selectQuery = name => createSelector(
-  createSelector(
-    state => state.wp.getIn(['requestsByName', name]),
-    state => state.wp.getIn(['requestsByQuery']),
-    (currentRequest, cache) => {
-      if (!currentRequest) return false;
-
-      const cacheID = currentRequest.get('cacheID');
-      const page = currentRequest.get('page');
-
-      if (cacheID) {
-        return currentRequest
-        .merge(cache.getIn([cacheID, page]))
-        .merge(cache.getIn([cacheID, 'pagination']));
+export const selectRequest = (requestData) => {
+  let requestIDSelector;
+  if (isString(requestData)) {
+    requestIDSelector = state => state.wp.getIn(['requestsByName', requestData]);
+  } else if (requestData.name) {
```

Comment:
Try and avoid nesting of condition when they can be considered separately. Also, in this case include a check for the `requestData` being supplied by the user (additionally ass a test for this, if you've not considered it).